specification/dimensions/commitmentdiscounttype.md

```diff
@@ -0,0 +1,39 @@
+# Commitment Discount Type
+
+A commitment-based discount is a contractual commitment for an amount of usage or spend throughout a specified term, in exchange for discounted unit pricing on that amount. The commitment may be based on quantities of resource units or monetary value, with various payment options and time frames.
```

Comment:
Ideally, it'd be great to clearly distance this definition from a "capacity reservation" which absolutely isn't a discount.

You could call out that capacity guarantees MAY be coupled with a commitment discount type (but not always).

Reply:
If making an update here, I assume we would have to make it elsewhere. I like "Commitment Discount *" not being mixed with on-demand capacity reservations in AWS (and the equivalents) since they don't apply here. Would we make a differentiation where "Commitment Discount *" columns only apply to cases where your commitment gets discounts and "Commitment *" columns getting used when a column must apply to either?

Reply:
I think that makes sense

Reply:
Do you think we need to explicitly call this out here?

Reply:
Hey @cnharris10,
Why do you think it can be confused with capacity reservation? Capacity reservation doesn't provide discount and can be cancelled at any time, so it is not really a commitment and doesn't fall into above definition. 
Another note, what is meant by word 'contractual' here? Is it really needed?

Reply:
I can see your point @yprikhodko and @flanakin. Ok if we don't make any reference to capacity reservations. 

Conceptually the only similarities are the fact that you have a used and unused portion that would need to be charged to the capacity reservation itself and the fact that there's a 'reservation' not a 'commitment' per se.

Also ok without 'contractual' in the description. It's no more contractual than any other usage or purchase with a provider.

Reply:
Simply, the fact that AWS and Azure uses "reservation" in "Capacity Reservation", GCP literally calls a capacity reservation a "Reservation", and AWS/Azure-like "reservations" are "commitments" (except BigQuery reservations are "reservations") in GCP makes differentiating these terms incredibly complex.

Just a line that says, ~"Commitments may be bundled with infrastructure guarantees but entirely disjoint from purchases that don't incur a fix term" would go a long way.

Reply:
I have removed the word "contractual" in the following files:

- commitmentdiscountname
- commitmentdiscounttype
- commitmentdiscountid